packages/adaptive_scaffold/lib/src/adaptive_layout.dart

```diff
@@ -0,0 +1,509 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'breakpoints.dart';
+import 'slot_layout.dart';
+import 'slot_layout_config.dart';
+
+const String _kPrimaryNavigationID = 'primaryNavigation';
```

Comment:
I think these would be better as a private enum, instead of as separate strings. They'd take up less space in release builds.

Reply:
The LayoutId from the custom builder delegate takes Strings. Is there some way to use an enum while fitting this use as well?

Reply:
You can use the enum names.

For instance, if you have an enum:

```dart
enum _LayoutSlot {
  startNavigation,
  endNavigation,
}
```

You can supply `_LayoutSlot.startNavigation.name` as the `LayoutId`. I guess it will get awkward if you have to do that a lot, though. This is an implementation detail, so it's up to you if you still want to convert them to enums.

Reply:
I think it makes sense to have it in an enum since it'll even make iterating over the ids easier if needed. I'll use this strategy. Thanks.